app/src/main/java/io/gnosis/safe/ui/safe/share/ShareSafeViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package io.gnosis.safe.ui.safe.share
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import androidx.annotation.ColorInt
+import androidx.lifecycle.viewModelScope
+import io.gnosis.data.models.Safe
+import io.gnosis.data.repositories.EnsRepository
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.transaction.NoSafeSelected
+import kotlinx.coroutines.withContext
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.svalinn.common.utils.QrCodeGenerator
+import timber.log.Timber
+import javax.inject.Inject
+
+class ShareSafeViewModel
+@Inject constructor(
+    private val safeRepository: SafeRepository,
+    private val ensRepository: EnsRepository,
+    private val qrCodeGenerator: QrCodeGenerator,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<ShareSafeState>(appDispatchers) {
+
+    override fun initialState(): ShareSafeState = ShareSafeState()
+
+    fun load() {
+        safeLaunch {
+            safeRepository.getActiveSafe()?.let { activeSafe ->
+                val ensName = runCatching { ensRepository.reverseResolve(activeSafe.address) }
+                    .onFailure { Timber.e(it) }
+                    .getOrNull()
+                val qrCode = runCatching {
+                    qrCodeGenerator.generateQrCodeSync(
+                        activeSafe.address.asEthereumAddressChecksumString(),
+                        512,
+                        512,
+                        Color.WHITE
+                    )
+                }.onFailure { Timber.e(it) }
```

Comment:
should we propagate qr code generation error to user and show a snackbar? Or failing silently would be enough?

Reply:
The legacy app failed silently, I think. We still log, but I think that the rest of the UI is still meaningful without a QR code.